extensions/control-plane/provision/provision-aws-s3/src/test/java/org/eclipse/edc/connector/provision/aws/s3/S3BucketProvisionerTest.java

```diff
@@ -100,10 +103,12 @@ void verify_basic_provision() {
             assertThat(secretToken.sessionToken()).isEqualTo("sessionToken");
         });
         verify(iamClient).putRolePolicy(isA(PutRolePolicyRequest.class));
+        verify(s3Client).createBucket(isA(CreateBucketRequest.class));
     }
 
     @Test
     void should_return_failed_future_on_error() {
+        when(s3Client.headBucket(isA(HeadBucketRequest.class))).thenReturn(failedFuture(new RuntimeException("any")));
```

Comment:
this call should return successful response, so it will be clear that the error that causes the failed response is caused by the `createBucket` call.

Reply:
oh, right, sorry